strapi/strapi
# enhancement!: validate input in content API create and update controllers

### What does it do?

- `coreController` `create` and `update` methods call `validateInput()`
- `validateInput` now throws on fields that don't exist on schema
- fixes bug in `validateInput` where only the first traversal method was being called

### Why is it needed?

To make it easier to debug content API `create` and `update` requests; we should not be silently removing anything from the users input data.

### How to test it?

API tests have been added

To test manually, create a content type and try sending various types of bad input to it (create and update endpoints), and you should receive a 400 bad request instead of a 200/201 success.

### Related issue(s)/PR(s)

DX-1136
